language/move-prover/docgen/src/docgen.rs

```diff
@@ -664,7 +691,179 @@ impl<'env> Docgen<'env> {
+
+        visited.insert(fun_id);
+        queue.push_back(fun_id);
+
+        while let Some(id) = queue.pop_front() {
+            let caller_env = self.env.get_function(id);
+            let caller_name = name_of(&caller_env);
+            let callee_list = caller_env.get_called_functions();
+
+            if fun_env.module_env.get_id() == caller_env.module_env.get_id() {
+                dot_src.push_str(&format!("    {}\n", caller_name));
+            } else {
+                let module_name = caller_env
+                    .module_env
+                    .get_name()
+                    .display(caller_env.module_env.symbol_pool());
+                dot_src.push_str(&format!("    subgraph cluster_{} {{\n", module_name));
+                dot_src.push_str(&format!("        label = \"{}\";\n", module_name));
+                dot_src.push_str(&format!(
+                    "        {}[label=\"{}\"]\n",
+                    caller_name,
+                    caller_env.get_simple_name_string()
+                ));
+                dot_src.push_str("    }\n");
+            }
+
+            for callee_id in callee_list.iter() {
+                let callee_env = self.env.get_function(*callee_id);
+                let callee_name = name_of(&callee_env);
+                dot_src.push_str(&format!("    {} -> {}\n", caller_name, callee_name));
+                if !visited.contains(callee_id) {
+                    visited.insert(*callee_id);
+                    queue.push_back(*callee_id);
+                }
+            }
+        }
+        dot_src.push_str("}\n");
+
+        let out_file_path = PathBuf::from(&self.options.output_directory)
+            .join("img")
+            .join(format!(
+                "{}_call_graph.svg",
+                fun_env.get_name_string().to_string().replace("::", "_")
+            ));
+
+        self.gen_svg_file(&out_file_path, &dot_src);
+    }
+
+    /// Generate a forward (or backward) dependency diagram (.svg) for the given module.
+    fn gen_dependency_diagram(&self, module_id: ModuleId, is_forward: bool) {
+        let module_env = self.env.get_module(module_id);
+        let module_name = module_env.get_name().display(module_env.symbol_pool());
+
+        let mut dot_src: String = String::from("digraph G {\n");
+        let mut visited: BTreeSet<ModuleId> = BTreeSet::new();
+        let mut queue: VecDeque<ModuleId> = VecDeque::new();
+
+        visited.insert(module_id);
+        queue.push_back(module_id);
+
+        while let Some(id) = queue.pop_front() {
+            let mod_env = self.env.get_module(id);
+            let mod_name = mod_env.get_name().display(mod_env.symbol_pool());
+            let dep_list = if is_forward {
+                mod_env.get_used_modules(false)
+            } else {
+                mod_env.get_using_modules(false)
+            };
+            dot_src.push_str(&format!("    {}\n", mod_name));
+            for dep_id in dep_list.iter() {
```

Comment:
Iterating over `dep_list.iter().filter(|dep_id| dep_id != id) { ....}` could make this cleaner and would be more "Rusty" as you shouldn't need the `if .. { continue }` below.

Reply:
Done.